pkgs/data/themes/sddm-sugar-dark/default.nix

```diff
@@ -0,0 +1,47 @@
+{ pkgs, lib, stdenvNoCC, themeConfig ? null }:
+
+let
+  name = "sugar-dark";
+  owner = "MarianArlt";
+
+  basePath = "$out/share/sddm/themes/${name}";
+  iniFormat = pkgs.formats.ini { };
+in
+
+stdenvNoCC.mkDerivation rec {
+  pname = "sddm-${name}";
+  version = "1.2";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "${owner}";
+    repo = "${pname}";
+    rev = "v${version}";
+    hash = "sha256-C3qB9hFUeuT5+Dos2zFj5SyQegnghpoFV9wHvE9VoD8=";
+  };
+
+  propagatedUserEnvPkgs = with pkgs.libsForQt5.qt5; [ qtgraphicaleffects ];
+
+  installPhase =
+    let
+      configFile = iniFormat.generate "" { General = themeConfig; };
+    in
+    ''
+      mkdir -p ${basePath}
+      cp -r $src/* ${basePath}
+    '' + lib.optionalString (themeConfig != null) ''
+      ln -sf ${configFile} ${basePath}/"theme.conf.user"
```

Comment:
```suggestion
      mkdir -p $out/share/sddm/themes/sugar-dark
      cp -r $src/* $out/share/sddm/themes/sugar-dark
    '' + lib.optionalString (themeConfig != null) ''
      ln -sf ${configFile} $out/share/sddm/themes/sugar-dark/theme.conf.user"
```

Reply:
I moved the base path into a narrower scope but I don't think getting rid of it is a good idea. There's just no point in spelling out the same path 3 times. If the path is ever changed we'd have to change 3 lines instead of one.